spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationNode.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.support.management.graph;
+
+import org.springframework.integration.support.context.NamedComponent;
+
+/**
+ * Base class for all endpoints.
+ *
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+public abstract class IntegrationNode {
+
+	private final int nodeId;
+
+	private final String name;
+
+	private final String output;
+
+	private final String input;
```

Comment:
M-m-m. Not sure in these properties here in the common abstraction.
For example `MessageChannelNode` can't have them at all. 
So, I'd like do not see them in the tree, even if it is `null`.